maestro-client/src/main/java/maestro/OutgoingRequest.kt

```diff
@@ -0,0 +1,42 @@
+package maestro
+
+import maestro.mockserver.MockEvent
+import maestro.utils.StringUtils.toRegexSafe
+
+data class OutgoingRequestRules(
+    val path: String? = null,
+    val headersPresent: List<String> = emptyList(),
+    val headersAndValues: Map<String, String> = emptyMap(),
+    val httpMethodIs: String? = null,
+    val requestBodyContains: String? = null,
+)
+
+object AssertOutgoingRequestService {
+
+    private val REGEX_OPTIONS = setOf(RegexOption.IGNORE_CASE, RegexOption.DOT_MATCHES_ALL, RegexOption.MULTILINE)
+
+    fun assert(events: List<MockEvent>, rules: OutgoingRequestRules): List<MockEvent> {
```

Comment:
Big nit: shall we use a different name since `assert` is a reserved word in Java?